modules/apps/transfer/types/forwarding_info.go

```diff
@@ -0,0 +1,30 @@
+package types
+
+import (
+	errorsmod "cosmossdk.io/errors"
+
+	host "github.com/cosmos/ibc-go/v8/modules/core/24-host"
+)
+
+const MaximumNumberOfForwardingHops = 64
+
+func (fi ForwardingInfo) Validate() error {
```

Comment:
Yeah, that would be nice! What about Hops? :)

Reply:
I added the constructor, and if we agree on it I will create a separate issue to use it elsewhere in the code as well - didn't want to pollute this PR too much.

Reply:
I considered that, but I looked at some of the other types like:
```go
func (d Denom) Validate() error {
...

func (t Trace) Validate() error {
...
```

But I agree that ValidateBasic is better to keep this consistent with msg validation. I will change this one to ValidateBasic, and maybe just make a separate PR for the others? WDYT?

Reply:
I am OK with either, but I think we should agree upon this (and maybe document the pattern?)

Reply:
@crodriguezvega If you are OK with keeping `Validate` for now we can discuss this and change the others if we prefer - but I guess it would be better to keep it similar to the other ones for now for consistency with the non-msg `Validate` names.

Reply:
+1 for `Validate` and explicitly not implementing `HasValidateBasic`